core/node/logs_bloom_backfill/src/lib.rs

```diff
@@ -0,0 +1,113 @@
+use std::time::Duration;
+
+use anyhow::Context;
+use tokio::sync::watch;
+use zksync_dal::{Connection, ConnectionPool, Core, CoreDal};
+use zksync_types::{block::build_bloom, BloomInput, L2BlockNumber};
+
+#[derive(Debug)]
+pub struct LogsBloomBackfill {
+    connection_pool: ConnectionPool<Core>,
+}
+
+impl LogsBloomBackfill {
+    pub fn new(connection_pool: ConnectionPool<Core>) -> Self {
+        Self { connection_pool }
+    }
+
+    async fn wait_for_l2_block_with_bloom(
+        connection: &mut Connection<'_, Core>,
+        stop_receiver: &mut watch::Receiver<bool>,
+    ) -> anyhow::Result<bool> {
```

Comment:
Bikeshedding: IMO, it would be more idiomatic to introduce an enum for successful results here:

```rust
enum BloomWaitOutcome {
    Ok,
    Canceled,
}
```